ci-operator/step-registry/enable-qe-catalogsource/disconnected/enable-qe-catalogsource-disconnected-commands.sh

```diff
@@ -460,27 +481,29 @@ echo "MIRROR_PROXY_REGISTRY: ${MIRROR_PROXY_REGISTRY}"
 
 set_CA_for_nodes
 # get cluster Major.Minor version
-ocp_version=$(oc version -o json | jq -r '.openshiftVersion' | cut -d '.' -f1,2)
-origin_index_image="quay.io/openshift-qe-optional-operators/aosqe-index:v${ocp_version}"
-mirror_index_image="${MIRROR_PROXY_REGISTRY_QUAY}/openshift-qe-optional-operators/aosqe-index:v${ocp_version}"
+#
+#ocp_version=$(oc version -o json | jq -r '.openshiftVersion' | cut -d '.' -f1,2)
+kube_major=$(oc version -o json |jq -r '.serverVersion.major')
+kube_minor=$(oc version -o json |jq -r '.serverVersion.minor')
+origin_index_image="quay.io/openshift-qe-optional-operators/aosqe-index:v${kube_major}.${kube_minor}"
+mirror_index_image="${MIRROR_PROXY_REGISTRY_QUAY}/openshift-qe-optional-operators/aosqe-index:v${kube_major}.${kube_minor}"
+echo "origin_index_image: ${origin_index_image}"
+echo "mirror_index_image: ${mirror_index_image}"
+
+check_olm_capability
+check_marketplace
+# No need to disable the default OperatorHub when marketplace disabled as default.
+if [ $marketplace -eq 0 ]; then
+    disable_default_catalogsource
+fi
 
 if [ $mirror -eq 1 ]; then
+    echo "Mirror operator images as cluster is C2S or SC2S"
     mirror_optional_images
 else
-    # no need to set auth for the MIRROR_REGISTRY_HOST
+    echo "set_cluster_auth as the cluster isn't c2s or SC2S"
     set_cluster_auth
 fi 
 
 create_settled_icsp
```

Comment:
This ICSP creation function should be run before the OLM and Marketplace checking(`check_olm_capability`) since some 2-level operators(such sample-operator) depends on it, see more: https://github.com/openshift/release/pull/45365